packages/front-end/components/Settings/NewDataSourceForm.tsx

```diff
@@ -0,0 +1,558 @@
+import {
+  FC,
+  useState,
+  useEffect,
+  ChangeEventHandler,
+  ReactElement,
+} from "react";
+import { useAuth } from "../../services/auth";
+import { DataSourceInterfaceWithParams } from "back-end/types/datasource";
+import AthenaForm from "./AthenaForm";
+import PostgresForm from "./PostgresForm";
+import GoogleAnalyticsForm from "./GoogleAnalyticsForm";
+import SnowflakeForm from "./SnowflakeForm";
+import BigQueryForm from "./BigQueryForm";
+import ClickHouseForm from "./ClickHouseForm";
+import MixpanelForm from "./MixpanelForm";
+import track from "../../services/track";
+import PrestoForm from "./PrestoForm";
+import MysqlForm from "./MysqlForm";
+import SelectField from "../Forms/SelectField";
+import { getInitialSettings } from "../../services/datasources";
+import PagedModal from "../Modal/PagedModal";
+import Page from "../Modal/Page";
+import { MdKeyboardArrowDown } from "react-icons/md";
+import {
+  eventSchemas,
+  dataSourceConnections,
+  eventSchema,
+} from "../../services/eventSchema";
+import Field from "../Forms/Field";
+import { useForm } from "react-hook-form";
+import LoadingOverlay from "../LoadingOverlay";
+import styles from "./NewDataSourceForm.module.scss";
+
+const NewDataSourceForm: FC<{
+  data: Partial<DataSourceInterfaceWithParams>;
+  existing: boolean;
+  source: string;
+  onCancel: () => void;
+  onSuccess: (id: string) => Promise<void>;
+}> = ({ data, onSuccess, onCancel, source, existing }) => {
+  const [dirty, setDirty] = useState(false);
+  const [step, setStep] = useState(0);
+  const [loading, setLoading] = useState(false);
+  const [schema, setSchema] = useState("");
+  const [showFullList, setShowFullList] = useState(false);
+  const [possibleTypes, setPossibleTypes] = useState(
+    dataSourceConnections.map((d) => d.type)
+  );
+
+  const [datasource, setDatasource] = useState<
+    Partial<DataSourceInterfaceWithParams>
+  >(null);
+  const [hasError, setHasError] = useState(false);
+  const DEFAULT_DATA_SOURCE: Partial<DataSourceInterfaceWithParams> = {
+    name: "My Datasource",
+    settings: {},
+  };
+  const form = useForm({
+    defaultValues: {
+      settings: data?.settings || DEFAULT_DATA_SOURCE.settings,
+    },
+  });
+  const schemasMap = new Map();
+  const dataSourcesMap = new Map();
+  eventSchemas.forEach((o) => {
+    schemasMap.set(o.value, o);
+  });
+  dataSourceConnections.forEach((d) => {
+    dataSourcesMap.set(d.type, d);
+  });
+  const selectedSchema = schemasMap.get(schema) || {
+    value: "custom",
+    label: "Custom",
+  };
+  useEffect(() => {
+    track("View New Datasource Form", {
```

Comment:
Keep all of these track event names the same as the old form (when applicable).  To differentiate them, add an event property like `newDatasourceForm: true`

Reply:
done